### DISCUSSION TITLE
Using image saved using `cosign save` in air gapped environment.

### DISCUSSION THREAD

**Discussion by rajeev-0** [sigstore/cosign#3646] [Q&A]
I could save the Image & signature using `cosign save`. then I can also verify signature using 'cosign verify --offline'. 
I am struggling with loading the verified image from file system (saved using `cosign save`) in docker runtime.
Is it possible to load the saved image directly using `docker` command ?

**Comment by adilburaksen** [comment]
`cosign save` stores the image in OCI directory layout, which `docker load` cannot read directly — `docker load` expects a Docker image archive (tar format), not an OCI directory.

Two options to load the saved image in Docker:

**Option 1: Push to a local registry, then pull**
```sh
# Start a local registry (one-time setup in the air-gapped env)
docker run -d -p 5000:5000 --name registry registry:2

# Load from the OCI directory into the local registry
cosign load --dir ./saved-image localhost:5000/myimage:tag

# Now use it normally with Docker
docker pull localhost:5000/myimage:tag
docker run localhost:5000/myimage:tag
```

**Option 2: Convert OCI dir → Docker archive with skopeo**
```sh
# Convert OCI directory to docker archive format
skopeo copy oci:./saved-image docker-archive:./myimage.tar:myimage:tag

# Load the tar into Docker
docker load < ./myimage.tar
```

`cosign load` also re-attaches the signature and attestations from the saved directory when pushing to the local registry, so you can still verify offline:
```sh
cosign verify --offline --certificate-identity=... --certificate-oidc-issuer=... localhost:5000/myimage:tag
```

Option 1 is simpler if you're setting up a permanent air-gapped environment. Option 2 is better for one-off transfers.
